docs-ref-conceptual/create-an-azure-service-principal-azure-cli.md

```diff
@@ -36,58 +36,79 @@ The `appId` and `tenant` keys appear in the output of `az ad sp create-for-rbac`
 
 When creating a service principal, you choose the type of sign-in authentication it uses. There are two types of authentication available for Azure service principals: password-based authentication, and certificate-based authentication.
 
-> [!NOTE]
->
-> If your account doesn't have permission to create a service principal, `az ad sp create-for-rbac` will return an error message containing
-> "Insufficient privileges to complete the operation." Contact your Azure Active Directory admin to create a service principal.
-
 > [!WARNING]
 > When you create an Azure service principal using the `az ad sp create-for-rbac` command, the output includes credentials that you must protect. Be sure that you do not include these credentials in your code or check the credentials into your source control. As an alternative, consider using [managed identities](/azure/active-directory/managed-identities-azure-resources/overview) if available to avoid the need to use credentials.
 >
-> We recommend using `Contributor` for the `--role` parameter at a minimum. To reduce your risk of a compromised service principal, assign a more specific role and narrow the scope to a resource or resource group. See [Steps to add a role assignment](/azure/role-based-access-control/role-assignments-steps) for more information.
+> To reduce your risk of a compromised service principal, assign a more specific role and narrow the scope to a resource or resource group. See [Steps to add a role assignment](/azure/role-based-access-control/role-assignments-steps) for more information.
 
 
 ### Password-based authentication
 
-Without any authentication parameters, password-based authentication is used with a random password created for you.
+With password-based authentication, a random password is created for you.  Along with a `--name`, you must specify a `--scope` as this value does not have a default.  If you prefer, you can set the role assignment later by using [az role assignment create](/cli/azure/role/assignment#az-role-assignment-create).
+
+```azurecli-interactive
+az ad sp create-for-rbac --name myServicePrincipalName \
+                         --role reader \
+                         --scope /subscriptions/mySubscriptionID/resourceGroups/myResourceGroupName
+```
 
-  ```azurecli-interactive
-  az ad sp create-for-rbac --name ServicePrincipalName --role Contributor
-  ```
+You can also create a service principal using variables.
 
-> [!IMPORTANT]
-> As of Azure CLI 2.0.68, the `--password` parameter to create a service principal with a user-defined password
-> is __no longer supported__ to prevent the accidental use of weak passwords.
+```azurecli-interactive
+let "randomIdentifier=$RANDOM*$RANDOM"  
+servicePrincipalName="msdocs-sp-$randomIdentifier"
```

Comment:
`az ad sp create-for-rbac` already has the functionality to generate a name using the current time stamp:

```
> az ad sp create-for-rbac
The output includes credentials that you must protect. Be sure that you do not include these credentials in your code or check the credentials into your source control. For more information, see https://aka.ms/azadsp-cli
{
  "appId": "79ff821d-5b68-4377-b6e2-f4d0714656bd",
  "displayName": "azure-cli-2022-02-28-06-54-15",
  "password": "...",
  "tenant": "54826b22-38d6-4fb2-bad9-b7b93a3e9c5a"
}
```

Reply:
@jiasli , correct, but only if they don't specify a name.  This variable example comes from how we are [standardizing Azure CLI samples](https://github.com/Azure-Samples/azure-cli-samples/blob/master/sql-database/backup-database/backup-database.sh) in Azure-samples/azure-cli-samples.